Go test with build tags doesn't run if the tag doesn't have a space

I have a typical Go project with build tags in tests. For some reason, tests without space after `//` in Go tag do not work:

```go
//go:build unit_test

package tests

import "testing"

func TestTarget2(t *testing.T) {
	t.Fail()
}
```

Outputs

```
ok      github.com/test/test (cached) [no tests to run]
[Process exited 0]
```

However

```go
// go:build unit_test

package tests

import "testing"

func TestTarget(t *testing.T) {
	t.Fail()
}
```

Runs as expected:
```
--- FAIL: TestTarget (0.00s)
FAIL
exit status 1
FAIL    github.com/test/test 0.182s

[Process exited 1]
```

I have tried it with all commands, `TestFile`, `TestNearest`, etc. Not sure what is the possible problem. For `vim-test` I use any standard configuration. 

vim-go plugin has this option:
`vim.g.go_build_tags = "unit_test,integration_test"`

And LSP is configured with `GOFLAGS = "-tags=integration_test,unit_test"`.

CLI run works fine:

```
$ go test -tags="unit_test" ./...                                                                                                                                                                          00:18:38
--- FAIL: TestTarget (0.00s)
--- FAIL: TestTarget2 (0.00s)
FAIL
FAIL    github.com/test/test 0.306s
FAIL
```

I will be happy to assist with a PR if someone points me to where the problem might be

--- a/autoload/test/go/gotest.vim
+++ b/autoload/test/go/gotest.vim
@@ -27,7 +27,7 @@ function! test#go#gotest#build_args(args) abort
   endif
   let tags = []
   let index = 1
-  let pattern = '^//\s*+build\s\+\(.\+\)'
+  let pattern = '^//\s*\%(go:build\|+build\)\s\+\(.\+\)'
   while index <= getbufinfo('%')[0]['linecount']
     let line = trim(getbufline('%', l:index)[0])
     if l:line =~# '^package '
@@ -36,8 +36,12 @@ function! test#go#gotest#build_args(args) abort
     let tag = substitute(line, l:pattern, '\1', '')
     if l:tag != l:line
       " replace OR tags with AND, since we are going to use all the tags anyway
-      let tag = substitute(l:tag, ' \+', ',', 'g')
-      call add(l:tags, l:tag)
+      let tag = substitute(l:tag, '\v\&\&|\|\||\(|\)', '', 'g')
+      for val in split(l:tag, '[, ]\+')
+        if index(l:tags, l:val) == -1
+          call add(l:tags, l:val)
+        endif
+      endfor
     endif
     let index += 1
   endwhile

--- a/spec/fixtures/gotest/build_tags_117_test.go
+++ b/spec/fixtures/gotest/build_tags_117_test.go
@@ -1,0 +1,11 @@
+// some comments are allowed before tags
+
+//go:build foo && hello && world && !bar && (red || black)
+
+package mypackage
+
+import "testing"
+
+func TestNumbers(*testing.T) {
+	// assertions
+}

--- a/spec/gotest_spec.vim
+++ b/spec/gotest_spec.vim
@@ -110,6 +110,17 @@ describe "GoTest"
     Expect g:test#last_command == 'go test -tags=foo,hello,world,!bar,red,black'
   end
 
+  it "runs tests in a file with only Go 1.17 build tags"
+    view +10 build_tags_117_test.go
+    TestNearest
+
+    Expect g:test#last_command == 'go test -tags=foo,hello,world,!bar,red,black -run ''TestNumbers$'' ./.'
+
+    TestFile
+
+    Expect g:test#last_command == 'go test -tags=foo,hello,world,!bar,red,black'
+  end
+
   it "runs test suite without tags"
     view +14 build_tags_test.go
     TestSuite

